pytx/pytx/common.py

```diff
@@ -0,0 +1,345 @@
+        :param dict_generator: Return a dictionary instead of an instantiated
+                               object.
+        :type dict_generator: bool
+        :param metadata: Get extra metadata in the response.
+        :type metadata: bool
+        :returns: Generator, dict, class
+        """
+
+        if isinstance(cls_or_self, type):
+            url = t.URL + id + '/'
+        else:
+            url = cls_or_self._DETAILS
+        if connection:
+            url = url + connection + '/'
+        params = init.Broker.build_get_parameters()
+        if isinstance(fields, basestring):
+            fields = fields.split(',')
+        if fields is not None and not isinstance(fields, list):
+            raise pytxValueError("fields must be a list")
+        if fields is not None:
+            params[t.FIELDS] = ','.join(f.strip() for f in fields)
+        if metadata:
+            params[t.METADATA] = 1
+        if full_response:
+            return init.Broker.get(url, params=params)
+        else:
+            if connection:
+                return init.Broker.get_generator(cls_or_self,
+                                                 url,
+                                                 t.NO_TOTAL,
+                                                 to_dict=dict_generator,
+                                                 params=params)
+            else:
+                if isinstance(cls_or_self, type):
+                    return init.Broker.get_new(cls_or_self,
+                                               init.Broker.get(url,
+                                               params=params))
+                else:
+                    cls_or_self.populate(init.Broker.get(url, params=params))
+
+    def save(self, url):
+        """
+        Save this object. If it is a new object, use self._fields to build the
+        POST and submit to the appropriate URL. If it is an update to an
+        existing object, use self._changed to only submit the modified
+        attributes in the POST.
+
+        :returns: dict (using json.loads())
+        """
+
+        if self._new:
+            params = dict(
+                (n, getattr(self, n)) for n in self._fields if n != c.ID
+            )
+            return self._post(self._URL, params=params)
+        else:
+            params = dict(
+                (n, getattr(self, n)) for n in self._changed if n != c.ID
+            )
+            return init.Broker.post(self._DETAILS, params=params)
+
+    def expire(self, timestamp):
+        """
+        Expire by setting the 'expired_on' timestamp.
+
+        :param timestamp: The timestamp to set for an expiration date.
+        :type timestamp: str
+        """
+
+        init.Broker.is_timestamp(timestamp)
+        self.set(ti.EXPIRED_ON, timestamp)
+        self.save()
+
+    def false_positive(self, object_id):
+        """
+        Mark an object as a false positive by setting the status to
+        NON_MALICIOUS.
```

Comment:
The convention we are trying to establish is that only whitelisted data should be marked as NON_MALICIOUS. When something is a false positive, it should not necessarily be added to the whitelist. As such, when reporting a false positive, it would be better to set the status to UNKNOWN.